cli/cmds/cluster_delete.go

```diff
@@ -55,6 +65,52 @@ func delete(clx *cli.Context) error {
 			Namespace: Namespace(),
 		},
 	}
+	// keep bootstrap secrets and tokens if --keep-data flag is passed
+	if keepData {
+		// skip removing tokenSecret
+		if err := RemoveOwnerReferenceFromSecret(ctx, k3kcluster.TokenSecretName(cluster.Name), ctrlClient, cluster); err != nil {
+			return err
+		}
+
+		// skip removing webhook secret
+		if err := RemoveOwnerReferenceFromSecret(ctx, agent.WebhookSecretName(cluster.Name), ctrlClient, cluster); err != nil {
+			return err
+		}
+	}
+
+	if err := ctrlClient.Delete(ctx, &cluster); err != nil {
+		return client.IgnoreNotFound(err)
+	}
```

Comment:
Maybe if not found we should still try to delete the data. Should we swap the deletions?

```go
if keepData {
    removeOwnerRefs
} else {
    deletePVCs
}
deleteCluster
```